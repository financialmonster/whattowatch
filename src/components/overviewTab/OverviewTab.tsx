import React, { FC, memo } from 'react';
import { Map, List } from 'immutable';

import { mapRatingToMark } from 'utils';

type TOverviewTabProps = {
    rating: string;
    scoresCount: number;
    description: string;
    director: string;
    starring: List<Map<string, any>>;
}

export const OverviewTab: FC<TOverviewTabProps> = memo(({rating, scoresCount, description, director, starring}) => {
    return (
        <>
            <div className="movie-rating">
                <div className="movie-rating__score">{rating}</div>
                <p className="movie-rating__meta">
                    <span className="movie-rating__level">{mapRatingToMark(+rating)}</span>
                    <span className="movie-rating__count">{`${scoresCount} ratings`}</span>
                </p>
            </div>
            <div className="movie-card__text">
                <p>{description}</p>
                <p className="movie-card__director">
                    <strong>{`Director: ${director}`}</strong>
                </p>
                <p className="movie-card__starring">
                    <strong>{`Starring: ${starring.join(`, `)}`}</strong>
                </p>
            </div>
        </>
    );
});
