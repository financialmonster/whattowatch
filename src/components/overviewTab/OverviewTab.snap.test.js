import React from 'react';
import renderer from 'react-test-renderer';
import {  List } from 'immutable';

import { OverviewTab } from './OverviewTab';

it(`OverviewTab: correctly rendered`, () => {
    const tree = renderer.create(
        <OverviewTab eleased={1} runTime={2} genre="Comedy" director="Testee" starring={List()} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
