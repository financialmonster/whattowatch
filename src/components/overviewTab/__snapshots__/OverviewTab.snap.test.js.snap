// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewTab: correctly rendered 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    />
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        User's mark
      </span>
      <span
        className="movie-rating__count"
      >
        undefined ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p />
    <p
      className="movie-card__director"
    >
      <strong>
        Director: Testee
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
      </strong>
    </p>
  </div>,
]
`;
